### PR TITLE
Add zookeeper_connect_timeout option

### PR DESCRIPTION
**Pull request by wangxiyuan**
Make zookeper client connection timeout configurable.

Related-Bug: theopenlab/openlab#218

### CODE REVIEW

File: openlabcmd/openlabcmd/zk.py [hunk 1]
```diff
@@ -84,19 +84,28 @@ def lost(self):
...
+                    "The config doesn't contain [ha]zookeeper_hosts option.")
+            try:
+                timeout = timeout or self.config.get('ha', 'zookeeper_timeout')
+                timeout = int(timeout)
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_timeout option.")
+            except ValueError:
+                raise exceptions.ClientError("zookeeper_timeout should be "
+                                             "ini-like format.")
```
Comment (huangtianhua) [inline openlabcmd/openlabcmd/zk.py]: int?

File: openlabcmd/openlabcmd/zk.py [hunk 2]
```diff
@@ -84,19 +84,28 @@ def lost(self):
...
+                hosts = hosts or self.config.get('ha', 'zookeeper_hosts')
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_hosts option.")
+            try:
+                timeout = timeout or self.config.get('ha', 'zookeeper_timeout')
+                timeout = int(timeout)
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_timeout option.")
```
Comment (liusheng) [inline openlabcmd/openlabcmd/zk.py]: why this is mandatory ？ does it has default value ?

File: etc/openlabcmd/openlab.conf.j2 [hunk 3]
```diff
@@ -3,3 +3,4 @@ cloud_conf = /etc/openstack/clouds.yaml
 
 [ha]
 zookeeper_hosts = {{ groups['zookeeper'] | map('extract', hostvars, 'ansible_host') | join(', ') }}
+zookeeper_timeout = 5
```
Comment (huangtianhua) [inline etc/openlabcmd/openlab.conf.j2]: Maybe named 'zookeeper_connect_timeout' is better.

File: openlabcmd/openlabcmd/zk.py (hunk 2 above)
Comment (liusheng) [reply openlabcmd/openlabcmd/zk.py]: if so, does the `config` object can define config options with type limit, required, and default value ?

File: openlabcmd/openlabcmd/zk.py [hunk 4]
```diff
@@ -84,19 +84,29 @@ def lost(self):
...
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_hosts option.")
+            try:
+                timeout = timeout or self.config.get(
+                    'ha', 'zookeeper_connect_timeout')
+                timeout = int(timeout)
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_timeout option.")
```
Comment (huangtianhua) [inline openlabcmd/openlabcmd/zk.py]: missed.

Comment (wangxiyuan) [reply openlabcmd/openlabcmd/zk.py]: Done

File: openlabcmd/openlabcmd/zk.py (hunk 2 above)
Comment (wangxiyuan) [reply openlabcmd/openlabcmd/zk.py]: 1. Done, default is 5.
2. I didn't find a way to do it.

File: openlabcmd/openlabcmd/zk.py [hunk 5]
```diff
@@ -84,19 +84,29 @@ def lost(self):
...
                                              'string should be provided.')
-            else:
-                try:
-                    hosts = self.config.get('ha', 'zookeeper_hosts')
-                except configparser.NoOptionError:
-                    raise exceptions.ClientError(
-                        "The config doesn't [ha]zookeeper_hosts option.")
+            try:
+                hosts = hosts or self.config.get('ha', 'zookeeper_hosts')
+            except configparser.NoOptionError:
```
Comment (Yikun) [inline openlabcmd/openlabcmd/zk.py]: nit(It's not the problem in this change): Should we also add NoSectionError in here.
[1] https://github.com/python/cpython/blob/45d75fa/Lib/configparser.py#L782

File: openlabcmd/openlabcmd/zk.py [hunk 6]
```diff
@@ -84,19 +84,29 @@ def lost(self):
...
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_hosts option.")
+
+        if not timeout:
+            timeout = self.config.get('ha', 'zookeeper_connect_timeout',
+                                      fallback=5)
+        try:
+            timeout = int(timeout)
```
Comment (Yikun) [inline openlabcmd/openlabcmd/zk.py:102]:
```python
>> int(-1)
-1
```
Maybe could add some limitation on positive value.

**Review by Yikun — CHANGES_REQUESTED** [review]

**Review by bzhaoopenstack — APPROVED** [review]
lgtm

File: openlabcmd/openlabcmd/zk.py (hunk 6 above)
Comment (bzhaoopenstack) [reply openlabcmd/openlabcmd/zk.py:102]: I saw this is fixed  in the following line.

**Pull request merged**